### ISSUE TITLE
Handle CentOS outdated kernel using CentOS's 'vault' repository

### ISSUE THREAD

**Issue by martin-langhoff** [dotless-de/vagrant-vbguest#223]
A perennial problem with CentOS is the removal of old `kernel-devel` packages. As discussed in #141 and #153 , it is a heavy-handed move to update the kernel.

However, CentOS makes available a 'vault' repo for every release. So for example, CentOS 6.x images contain `/etc/yum.repos.d/CentOS-Vault.repo` which defines a bunch of repos with the format `C6.x-$section`. 

Knowing this, we could try the equivalent of the following bash code:

```
EL_VERSION=$(grep -oP '\d+\.\d+' /etc/centos-release) 
yum install --enablerepo=C${EL_VERSION}-base  --enablerepo=C${EL_VERSION}-updates kernel-devel
```

This isn't perfect -- some popular CentOS images I've seen in the wild don't have the vault repos defined for their own version; or their kernel is from an earlier version. This would be slower, and have a better chance:

```
yum install --enablerepo=C*-base  --enablerepo=C*-updates kernel-devel
```

In any case, I'd do this _after_ the initial attempt failed. Looking at the code in redhat.rb, it might make sense to separate the kernel dependency from the others, so as to install the other deps first, then try the variations on kernel-devel installation.

Here's some docs on Vault.
http://wiki.r1soft.com/display/kb3/Finding+Old+kernel-devel+Packages+For+CentOS

Finally, I'll note that this does not affect RHEL proper, as old kernel/packages remain in the RHEL repos forever as a matter of RHEL policy.

**Comment by oliver-dungey** [comment]
I know there are concerns about implementing this sort of thing but I've just lost another afternoon trying to resolve this issue after updating things, and I've been here before many times. Would love it to just sort itself out - I'm sure many newbies just give up and walk away.

**Comment by fnordfish** [comment]
Hi @oliver-dungey - would you mind testing my try of implementing @martin-langhoff's suggestion over in #231?

**Comment by martin-langhoff** [comment]
@fnordfish - what CentOS box are you testing this on? I know some have busted Vault repo definitions (to make things even worse :-/ ) .

Does the yum command I suggested work? You could just try enabling 'all' the vault repos.

**Comment by oliver-dungey** [comment]
@fnordfish boxcutter/centos67, open to suggestions that are better - just looking for a nice slim standard install, no ui or any of that clutter.

I'll test it out on Monday and give you some details of the results.

**Comment by oliver-dungey** [comment]
@fnordfish I ran the first variant and it failed with the error that @martin-langhoff said might happen: `Error getting repository data for C6.7-base, repository not found`.

After running `yum install --enablerepo=C*-base  --enablerepo=C*-updates kernel-devel` I performed a `vagrant reload` which correctly identified that I needed to upgrade from 5.0.20->5.0.26. But during the install I noticed that it didn't download the new ISO but used an old 5.0.4 version that was stored locally (`C:\Program Files\Oracle\VirtualBox\VBoxGuestAdditions.iso`), I checked the date time on the file and it was the same as the 5.0.26 install, strange.

I the deleted the ISO file and ran reload again and see a message again saying `Copy iso file C:\Program Files\Oracle\VirtualBox\VBoxGuestAdditions.iso into the box /tmp/VBoxGuestAdditions.iso` ... but that file does not exist locally?!?! And the message below it says installing guest additions 5.0.4.

It looks to me like it is downloading the wrong ISO so it doesn't really matter what the state of the kernel headers is.
